starlette/middleware/gzip.py

```diff
@@ -48,6 +49,14 @@ async def send_with_gzip(self, message: Message) -> None:
             # Don't send the initial message until we've determined how to
             # modify the outgoing headers correctly.
             self.initial_message = message
+            self.content_encoding_set = any(
+                header == b"content-encoding"
+                for header, _ in message.get("headers", [])
+            )
```

Comment:
We have a preference to use the wrappers `Headers`, and `MutableHeaders`. 👀 
```suggestion
            headers = Headers(raw=self.initial_message["headers"])
            self.content_encoding_set = b"content-encoding" in headers
```